### PR TITLE
Support Joomla 4

### PR DESCRIPTION
**Pull request by bastiaandewaele**
## Description

This PR upgrades our code to be compatible with Joomla 4. 

- Made the code compatible with Joomla 4
- Added support for creating categories (new feature)
- Changed the license to GPL-3
- Changed the readme
  -  FAQ: custom fields requires the permissions to be changed to public

Because of a lot of code got refactored, our plugin no longer supports Joomla 3. Supporting both versions would have added to much complexity. 

To keep supporting both version I have added these branches.

- [3.X](https://github.com/Story-Chief/joomla-component-storychief/tree/3.x) -> Joomla 3
- [4.X](https://github.com/Story-Chief/joomla-component-storychief/tree/4.x) -> Joomla 4
- [develop](https://github.com/Story-Chief/joomla-component-storychief/tree/develop) -> Joomla 4

Once this PR is approved and merged, the master branch will also become **Joomla 4**. But we will still have our branch [3.X](https://github.com/Story-Chief/joomla-component-storychief/tree/3.x) to pushing bug-fixes to Joomla 3.

## Testing notes

Download the source code and install it in a Joomla 4 environment. 

- Connect channel in StoryChief
- Publish
- Update
- Unpublish
- ... 

## Release notes

The Joomla destination detail / edit page contains a link to (button "view plugin"):

https://github.com/Story-Chief/joomla-component-storychief/tree/v1.0.13

This needs to be changed to:

https://github.com/Story-Chief/joomla-component-storychief/releases

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,20 +1,23 @@
...
 - Imports text and images from your StoryChief story
 - Supports custom fields
 - Support multi-language
-- Supports tags
+- Supports categories, tags
 
 ## HOW IT WORKS
+
 1. [Register](https://app.storychief.io/register) on StoryChief
 2. Add a 'Custom Website' channel
```
Comment (gregory-claeyssens) [inline README.md:22]: Think this is outdated

Comment (gregory-claeyssens) [inline README.md]: This is outdated

File: README.md [hunk 2]
```diff
@@ -23,18 +26,29 @@ This component:
 6. Publish from Story Chief to your Joomla! website
 
 ## REQUIREMENTS
-- This component requires a Story Chief account.
+- This component requires a StoryChief account.
     - Not a Story Chief user yet? [Sign up](https://app.storychief.io/register) for free!
-- Joomla! 3.9 or higher
-- PHP version 7.0 or higher
+- Joomla! 4.0 or higher
```
Comment (gregory-claeyssens) [inline README.md:31]: Add instruction or reference to V3

**Comment by gregory-claeyssens** [comment]
Issue when installing (screenshot). It does install the component properly though so it's just a warning
Screenshot 2022-07-20 135008

**Comment by gregory-claeyssens** [comment]
I suggest renaming the 'field mapping' to 'custom field mapping'
Screenshot 2022-07-20 135018

**Comment by gregory-claeyssens** [comment]
Notices when publishing (Postman collection>publish event)
image
image
image

**Comment by gregory-claeyssens** [comment]
FYI, in the end I'll delete master and develop branch and only keep the 3.x and 4.x branches.
Those branches will serve as the develop branch for each specific branch. Releases will be used for production ready plugin versions

**Review by gregory-claeyssens — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (bastiaandewaele) [reply README.md]: See commit 1273483d23453698b66afe9ddcad1c1d6b8ff771

**Comment by bastiaandewaele** [comment]
> I suggest renaming the 'field mapping' to 'custom field mapping' Screenshot 2022-07-20 135018

Please see commit e00d0cb717c3926a18f35686496107143b5d621e

File: README.md (hunk 2 above)
Comment (bastiaandewaele) [reply README.md:31]: See commit 44e1b86730c5ef017daf2830456a90cc41e32951

**Comment by bastiaandewaele** [comment]
> Issue when installing (screenshot). It does install the component properly though so it's just a warning Screenshot 2022-07-20 135008

See commit 65a594fb46e602dd0cff9aa7dadff9a594da6c4e

**Comment by bastiaandewaele** [comment]
The PHP-errors should be fixed now for PHP 8.

**Review by gregory-claeyssens — APPROVED** [review]

**Pull request merged**